q2_longitudinal/_longitudinal.py

```diff
@@ -235,6 +237,46 @@ def linear_mixed_effects(output_dir: str, metadata: qiime2.Metadata,
                plot_name='Regression scatterplots', residuals=res)
 
 
+def anova(output_dir: str,
+          metadata: qiime2.Metadata,
+          formula: str,
+          sstype: int = 2) -> None:
+
+    # Grab metric and covariate names from formula
+    metric, group_columns = _parse_formula(formula)
+    columns = [metric] + list(group_columns)
+
+    # Validate formula (columns are in metadata, etc)
+    for col in columns:
+        metadata.get_column(col)
+    metadata = metadata.to_dataframe()[columns].dropna()
+
+    # Run anova
+    lm = ols(formula, metadata).fit()
+    results = pd.DataFrame(sm.stats.anova_lm(lm, typ=sstype)).fillna('')
+    results.to_csv(os.path.join(output_dir, 'anova.tsv'), sep='\t')
+
+    # Run pairwise t-tests with multiple test correction
+    pairwise_tests = pd.DataFrame()
+    for group in group_columns:
+        # only run on categorical columns — numeric columns raise error
+        if metadata[group].dtype == 'O':
```

Comment:
How about we use the [`qiime2.Metadata` API](https://dev.qiime2.org/latest/api-reference/metadata/#qiime2.metadata.Metadata.filter_columns) to preemptively identify the `Categorical` columns?